types/api_addn_names.pp

```diff
@@ -0,0 +1,7 @@
+# a type to describe api_addn_names
+type K8s::Api_addn_names = Array[
```

Comment:
This should probably not be a different type, but rather have just a `K8s::Addn_names` type and then use `Optional[K8s::Addn_names]` for where it's optional.

Though with something like `K8s::SAN`/etc instead, as above